src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
+  //when it's attached
+  var observer = null;
+  if (!document.contains(el)) {
```

Comment:
TIL that `document.contains` is not a standard thing:  http://stackoverflow.com/questions/20557115/mocha-casperjs-headless-testing-document-contains-undefined

but `if (!document.body.contains(el)) {` works fine as suggested in that SO answer. This should fix the issue with the headless tests.